Feature/wincher upgrade callouts

KaisZaoualiWincher opened the pull request:
### Testing steps:
- Apply the following patch to the [api](https://gitlab.int.wincher.com/wincher/solution/-/merge_requests/602) (to hit local /yoast/upgrade-campaign, not merged to prod yet). The current patch simulates a paying user. To simulate a free user you can remove changes made to the Yoast controller. 
[api.patch](https://github.com/Yoast/wordpress-seo/files/11206094/api.patch)
- Apply the following patch to the yoast branch (to use the local campaign endpoint).  [yoast.patch](https://github.com/Yoast/wordpress-seo/files/11206100/yoast.patch)
- We need to test a paying user + tracking >= 80% of keywords: no discount mentioned, only upgrade link, and we display a callout with the title
- We need to test a free user  + tracking >= 80% of keywords: callout + discount section to be displayed
- We need to test a user  + tracking < 80% of keywords: no callout.
- Callouts should be displayed in the SEO performance report in the dashboard, and the edit post page (both in page and modal)

### Note
- Based on https://app.shortcut.com/wincher/story/6212/yoast-dashboard-connect-button (PR: https://github.com/Yoast/wordpress-seo/pull/20107)
- Yoast repo PR (DRAFT): https://github.com/Yoast/wordpress-seo/pull/20144
- Yoast gonna need to use https://app.wincher.com/websites?plans=true&utm_medium=plugin&utm_source=yoast&referer=yoast&partner=yoast for https://yoa.st/wincher-upgrade

### Evidence
- paid user + 80% tracking

![image](https://user-images.githubusercontent.com/129404933/231321069-36a0623f-bfca-4080-b00f-2d718d439728.png)

- free account + 80% tracking / modal

![image](https://user-images.githubusercontent.com/129404933/231321124-905b9b6b-8cc5-445c-b815-e4cff7a6b170.png)

- free account + 80% tracking / edit page

![image](https://user-images.githubusercontent.com/129404933/231321224-371a3759-3247-4d6c-9803-ee5422b30c2a.png)

- free account + 80% tracking / dashboard page

![image](https://user-images.githubusercontent.com/129404933/231321271-12b4bacc-e218-4ac4-8374-c014f40483e6.png)

jrfnl committed:
CS/QA: remove duplicate `global` variable import

Variable is already being imported earlier on in the same function.

jrfnl committed:
CS/QA: no need to unset non-existent vars

jrfnl committed:
PHP 8.2 | Indexable_*_Presentation: declare the property in the trait instead

jrfnl committed:
GH Actions: fix minor spelling snafu

jrfnl committed:
Integration tests: update script to install WP

Re-synced with the script from the WP-CLI scaffold-command repo.

Ref: https://github.com/wp-cli/scaffold-command/blob/eb8d71618de1e34264991109f91a8d8247601fb2/templates/install-wp-tests.sh

enricobattocchi committed:
Deprecate the integration to display a notification when Premium is not up-to-date

enricobattocchi committed:
Upgrade deprecation version

enricobattocchi committed:
Delete the test file once again

KaisZaoualiWincher committed:
Add upgrade callout to seo performance report

KaisZaoualiWincher committed:
Fetch and use the wincher upgrade discount

KaisZaoualiWincher committed:
Remove free account from paid user's callout

KaisZaoualiWincher committed:
Add upgrade link in wincher upgrade callout to paying users

skrivanos commented:
Bunch of failing code style checks, please fix them

skrivanos commented:
Is it possible to hide the close button on the callouts? It's sorta annoying UX when you close it, then see it again next time you open the modal.

skrivanos commented on packages/js/src/components/modals/WincherUpgradeCallout.js:
```diff
@@ -0,0 +1,257 @@
+		return null;
+	}
+
+	const isFreeAccount = Boolean( upgradeCampaign?.discount );
```
This is confusing and deserves a comment

KaisZaoualiWincher commented:
> Is it possible to hide the close button on the callouts? It's sorta annoying UX when you close it, then see it again next time you open the modal.

Do you mean remove it completely, or track it has been closed in current session (via redux)?

skrivanos commented:
> > Is it possible to hide the close button on the callouts? It's sorta annoying UX when you close it, then see it again next time you open the modal.
> 
> Do you mean remove it completely, or track it has been closed in current session (via redux)?

Was it in the design? If not I'd just remove it.

KaisZaoualiWincher commented:
Yes, it was in the design, but only for the post editor.

skrivanos commented:
> Yes, it was in the design, but only for the post editor.

I think we can just remove it for now, but maybe do it as a separate commit so we can revert if it's commented on later.

skrivanos commented on src/actions/wincher/wincher-account-action.php:
```diff
@@ -63,4 +64,26 @@ public function check_limit() {
+	public function get_upgrade_campaign() {
+		try {
+			$result = $this->client->get( self::UPGRADE_CAMPAIGN_URL );
+			$discount = $result['value'];
```
We don't support AMOUNT discounts here. Maybe that's fine but we should then ignore amount discounts returned from the API just in case. 

However, you need to return and use the number of months for the discount since that might change.

KaisZaoualiWincher replied:
Support of months added.

- Months returned by the API:
![image](https://user-images.githubusercontent.com/129404933/231828885-54279596-4bbc-4dcb-b62d-9b3947a0c2e5.png)

- No campaign:
![image](https://user-images.githubusercontent.com/129404933/231829996-a1091c7f-7f21-4707-8ddd-9a076aa6f67b.png)

- Amount-type campaign:
![image](https://user-images.githubusercontent.com/129404933/231830568-88619070-c50f-4469-ae88-534e43757cf5.png)

- Same for `months = null` and `months = 0`

KaisZaoualiWincher committed:
Fix linting

KaisZaoualiWincher committed:
Remove close upgrade callout icon and related action in Wincher SEO performance

KaisZaoualiWincher committed:
Add months support to the wincher upgrade campaign and use only rate type campaigns.

KaisZaoualiWincher committed:
Add comment to free account check in wincher upgrade callout.

vraja-pro committed:
Merged trunk into the branch and updated gitignore

pls78 committed:
Merge pull request #20069 from Yoast/19747-add-more-unit-tests-for-the-19388-pr-1

Tests for Indexable_Link_Builder

leonidasmi committed:
Detect whether a cron indexation should be scheduled only when on the right pages and in a GET request

thijsoo committed:
Adds a grouping array around the new indexable tracking info.

vraja-pro committed:
Added key to taxonomy submenu

thijsoo committed:
Update tests/unit/analytics/application/to-be-cleaned-indexables-collector-test.php

Co-authored-by: Paolo L. Scala <68744851+pls78@users.noreply.github.com>

pls78 committed:
Merge pull request #20259 from Yoast/20258-update-tracking-to-better-group-indexable-data

Adds a grouping array around the new indexable tracking info.

Commit:
Bump version to 20.7-RC8

pls78 committed:
Merge branch 'release/20.7' into trunk

leonidasmi committed:
Detect whether a cron indexation should be scheduled also when queries have already ran

enricobattocchi committed:
Merge pull request #20261 from Yoast/19863-warning-encountered-two-children-with-the-same-key

Added key to taxonomy submenu items

leonidasmi committed:
Add more unit tests

thijsoo committed:
Removing new tracking.

Commit:
Bump version to 20.7-RC9

KaisZaoualiWincher committed:
Add network error handling on Wincher connect

jrfnl committed:
Composer: update test dependencies

Refs:
* https://github.com/Yoast/PHPUnit-Polyfills/releases/tag/1.0.5
* https://github.com/sebastianbergmann/diff/blob/3.0.4/ChangeLog.md

jrfnl committed:
Dependabot: add reviewer config

The Dependabot PRs can have a tendency to stay open for a long time, while for the GH Actions and Composer dependency related ones I keep an eye on all used action runner releases anyway.

However, I don't "watch" all Yoast repos, so adding this configuration will ensure that I get notified about Dependabot PRs for this repo (and can merge it if appropriate or close it otherwise).

jrfnl committed:
CS/QA: lower thresholds

... to match the current reality to prevent new issues coming in.

jrfnl committed:
Tests: `TestCase` classes should be abstract

vraja-pro committed:
Fixed position of adding link popover

vraja-pro committed:
Added jsdoc comment to remove es lint warning

vraja-pro committed:
Added prop types and remove eslint warning

vraja-pro committed:
Reduced complexity by adding method isLinkNoFollow

leonidasmi committed:
Revert "Merge pull request #20005 from Yoast/19950-revisit-background-indexing-with-wp-cron"

This reverts commit 0615d631629d4baaa1718f85cc6ff6474d2026e0, reversing
changes made to 95bf91ec5b65559422b492f2e459ab66c88987d2.

vraja-pro committed:
Reduced complexity with method isToggleSettings

vraja-pro committed:
refactored isLinkNoFollow

vraja-pro committed:
Reduced complexity with isSponsored

vraja-pro committed:
Reduced complexity with didToggleSettingForNewLink

vraja-pro committed:
Reduced complexity with actionCompleteMessage

vraja-pro committed:
Reduced complexity with getNewText

vraja-pro committed:
Reduced complexity with shouldInsertLink

vraja-pro committed:
Removed complexity of actionCompleteMessage

vraja-pro committed:
Reduced complexity with validateLinkId

vraja-pro committed:
change onChangeLink to callback function to be passed as prop in LinkControl

vraja-pro committed:
reduced js warnings threshold

pls78 committed:
Add release text for 20.7

igorschoester committed:
Copy KeywordInput over from yoast-components

Commit:
Improve documentation

pls78 committed:
Merge pull request #20268 from Yoast/revert/cron-indexation

Reverts Cron indexation

jrfnl committed:
Merge pull request #20272 from Yoast/JRF/composer-update-dependencies

Composer: update test dependencies

jrfnl committed:
Merge pull request #20273 from Yoast/JRF/integration-tests-update-script

Integration tests: update script to install WP

jrfnl committed:
Merge pull request #20274 from Yoast/JRF/ghactions-minor-fix

GH Actions: fix minor spelling snafu

jrfnl committed:
Merge pull request #20275 from Yoast/JRF/tests-fix-broken-testcase

Tests: `TestCase` classes should be abstract

jrfnl committed:
Merge pull request #20276 from Yoast/JRF/dependabot-set-reviewer

Dependabot: add reviewer config

thijsoo committed:
Merge branch 'release/20.7' of github.com:Yoast/wordpress-seo into trunk

thijsoo committed:
Re-enable indexing.

hdvos committed:
make comment at spec for img clearer

hdvos committed:
remove commented out code in getTextElementPositionsSpec.js

hdvos committed:
small refactoring of getTextElementPositions.js based CR.

hdvos committed:
add documentation to findAllInTree.js

Commit:
Bump version to 20.7

Commit:
Merge branch 'release/20.7'

Commit:
Bump version to 20.7 on free

hdvos committed:
pass recurseFoundNodes when recursing

hdvos committed:
add spec for when recurseFoundNodes is true in findAllInTreeSpec.js

leonidasmi committed:
Merge branch 'main' of github.com:Yoast/wordpress-seo into trunk

vraja-pro committed:
Merge pull request #20278 from Yoast/20277-re-enabeling-the-indexable-tracking

Re-enable indexing.

vraja-pro committed:
updated class path for Abstract_Link_Indexing_Action_Test

vraja-pro committed:
Removed unused covers tag from  Aioseo_Cleanup_Action_Test

Commit:
Add changelog

Commit:
Bump version to 20.8-RC1

Commit:
Improve changelog

pls78 committed:
Merge pull request #20270 from Yoast/20015-difficulty-in-adding-a-link-in-the-editor-as-the-box-positioning-is-off

20015 difficulty in adding a link in the editor as the box positioning is off

igorschoester committed:
Improve AnalysisFields safety

igorschoester committed:
Add tests for AnalysisFields

igorschoester committed:
Copy contentAnalysis redux layer

* instead of using yoast-components
* add tests for the inclusive language reducer

jrfnl committed:
CS/QA: remove unused import `use` statements

jrfnl committed:
CS/QA: remove unnecessary import `use` statements

No need to import a class which is in the same namespace as the current class.

jrfnl committed:
CS/QA: alphabetically order import `use` statements

jrfnl committed:
CS/QA: import all used classes

vraja-pro committed:
Fixed Typo in for get method for twitter helpers surface

jrfnl committed:
CS/QA: use unqualified name for imported classes

vraja-pro committed:
Fixed typo for get method in schema helpers surface test

vraja-pro committed:
Fixed type in covers for get method in open graph helpers surface

vraja-pro committed:
Fixed typo in covers for get method in helpers surface

jrfnl committed:
CS/QA: don't use FQN function calls in non-namespaced files

vraja-pro committed:
removed covers of a mock in permalink test

vraja-pro committed:
removed covers for mocks in woocommerce integration test

vraja-pro committed:
Fixed covers for the tested method in settings integration test

vraja-pro committed:
Fixed covers in Robots_Txt_Integration_Test

* Removed change_default_robots since it is not a method in the tested class.
* fixed typo for maybe_add_xml_sitemap

vraja-pro committed:
Fixed Cleanup_Integration_Test covers and removed unused imports

* Removed unused imports
* Removed covers for mocks

vraja-pro committed:
Removed covers in Initializers\Woocommerce_Test

There is no construct method in the class.

vraja-pro committed:
Fixed path for Crawl_Cleanup_Helper in Crawl_Cleanup_Helper_Test

vraja-pro committed:
Updated path for Aioseo_Validation_Exception in Tests

vraja-pro committed:
Removed test for construct in Front_End_Conditional_Test

There is no construct method in the class.

vraja-pro committed:
Fixed path in covers for the tested method

vraja-pro committed:
Fixed path for tested abstract class in Post_Link_Indexing_Action_Test

vraja-pro committed:
Removed covers for get_post_types from Indexable_Post_Indexation_Action_Test

No such method in the tested class.

igorschoester committed:
Fix score elements for terms

By using different ID depending on isPost

igorschoester committed:
Switch to using the score actions from Free

Due to the previous fix of the AnalysisFields side-effects, we can now safely use the actions from Free instead of yoast-components

igorschoester committed:
Switch to using the content analysis reducer from Free

jrfnl committed:
CS/QA: use FQN constants in namespaced files

igorschoester committed:
Fix contentAnalysis tests

* use the new default IDs
* remove the created elements in the test, to not influence other tests

igorschoester committed:
Fix block editor test

Do not mock the combineReducers from WP data

jrfnl committed:
CS/QA: use FQN function calls in namespaced files

jrfnl committed:
Tests: fix incorrect test class name

jrfnl committed:
Tests: `set_up()` methods cannot "cover" anything

They should "set up" the environment for the test. The code in the setup should not be regarded as part of the test itself.

igorschoester committed:
Remove already unused yoast-components CSS

As the file suggests it was for the configuration wizard, which was replaced by the first time configuration

jrfnl committed:
CS/QA: static closures should be declared as static

igorschoester committed:
Remove the indirect import from yoast-components

With the style that is in that file

igorschoester committed:
Remove empty CSS file

jrfnl committed:
CS/QA: use the appropriate PHP native function

No need to use `array_map()` with a closure when there is a PHP native function to do exactly the same.

Ref: https://www.php.net/manual/en/function.array-column.php

jrfnl committed:
CS/QA: use `self` when refering to current class

jrfnl committed:
CS: various minor fixes

Mostly whitespace fixes and operator placements.

jrfnl committed:
CS/QA: import namespace, not function

jrfnl committed:
Tests: fix test not running due to duplicate array key in data provider

vraja-pro committed:
Add a pattern to add_disallow_search_to_robots

jrfnl committed:
CS/QA: match parameter name from WP Core

... to prevent (future) issues with the use of function calls with named parameters.

Ref: https://developer.wordpress.org/reference/functions/_deprecated_function/

jrfnl committed:
CS/QA: remove `global` statement for unused var

jrfnl committed:
CS/QA: don't use reserved keywords as param names

jrfnl committed:
CS/QA/Tests: don't use reserved keywords as param names

marinakoleva committed:
merges trunk

vraja-pro committed:
Updated unit test for add_disallow_search_to_robots

jrfnl committed:
CS/QA: fix inconsistent return types

These functions had a documented return type, which was previously adhered to, but now, they suddenly had an extra return type of `null`.

Fixed now.

jrfnl committed:
Docs: remove redundant `@throws` tag

Follow up on 19948, commit c5a257c1fe03cd5ba0aff28614d5670b5be61200

jrfnl committed:
Docs: use consistent tag order

jrfnl committed:
Docs: fix incorrect `@param` tag alignment

jrfnl committed:
CS/QA: remove unnecessary reference

KaisZaoualiWincher committed:
Fix Wincher performance alerts' margin bottom

jrfnl committed:
PHPCS: update ignore annotation

... with a reference to the upstream issue which should fix it (after which the annotation should be removed).

jrfnl committed:
CS/QA: fix file name of test class

... which was preventing this test from running.

Note: now the file name is fixed and the tests run, an error can be seen.

jrfnl committed:
SEO_Links_Repository: bug fix

If the passed `$indexable_ids`  array would be empty, a database error would occur:
```
WordPress database error:
[You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ") GROUP BY `target_indexable_id`" at line 1]

SELECT COUNT( id ) AS `incoming`, `target_indexable_id` FROM `wptests_yoast_seo_links` WHERE `target_indexable_id` IN () GROUP BY `target_indexable_id`
```

This small tweak fixes the bug.

jrfnl committed:
Tests: add missing assertion [1]

Both the `$expected` variable, as well as the `$actual` variable were unused, which gives me the impression the intention was to test that those are the same.

jrfnl committed:
Tests: sync `Addon_Manager_Double` with parent class

The `Addon_Manager::convert_subscription_to_plugin()` parent method received a new `$plugin_file` parameter over a year ago and while the parameter _was_ added to the `Addon_Manager_Double::convert_subscription_to_plugin()`, the parameter was not passed to the parent class, nor documented.

This fixes it and updates the associated test expectation.

jrfnl committed:
Tests: add missing assertion [2]

The `$expected` parameter received from the data provider was unused, which gives me the impression the intention was to also test the function call result.

jrfnl committed:
CS/QA: remove some unused variable assignments

The results of a function call do not need to be assigned to a variable, unless that variable will be used later on.

Along the same lines, when looping through an array, the key of the array item does not need to be assigned unless it will be used.

jrfnl committed:
CS/QA: remove unused variables

Variables which are not being used have no business existing. All of these are variables which are being assigned a value, but are subsequently never used.

jrfnl committed:
Tests: add missing assertion [3]

In each of these tests, an `$expected_result` variable is being set, but subsequently never used, which gives me the impression the intention was to also test the function call result.

jrfnl committed:
CS/QA: don't pass parameter which doesn't exist

Commit 47baef5bed4302c4bb76212339d327d462cf82fa removed support for the `$is_network` parameter, but there were still two method calls within the class passing that parameter.

Fixed now.

jrfnl committed:
Docs: minor formatting fixes

jrfnl committed:
Docs: property `@var` tags should not include the property name

jrfnl committed:
Docs: remove annotation which shouldn't be used

The `@access` annotation should only ever be used for functions declared in the global namespace where visibility modifiers are not available.

jrfnl committed:
Docs: use short form type keywords

jrfnl committed:
Docs: fix invalid `@return` tag descriptions

PHP does not return a named entity, so variable names have no place in `@return` tags.
Either provide a proper description or leave the description out.

jrfnl committed:
PHPCS: fix placement of ignore annotation

jrfnl committed:
Docs: fix capitalization and punctuation

... by actually describing the parameter.

jrfnl committed:
Merge pull request #20287 from Yoast/JRF/CSQA/fix-import-use-statements

CS/QA: fix up import `use` statements

jrfnl committed:
Merge pull request #20288 from Yoast/JRF/Tests/fix-tests-which-werent-being-run

Tests: fix some tests which werent being run

jrfnl committed:
Merge pull request #20289 from Yoast/JRF/Tests/more-covers-fixes

Tests: `set_up()` methods cannot "cover" anything

jrfnl committed:
Merge pull request #20290 from Yoast/JRF/Tests/fix-test-double

Tests: sync `Addon_Manager_Double` with parent class

jrfnl committed:
Merge pull request #20291 from Yoast/JRF/QA/parameter-name-fixes

QA: various parameter name fixes

jrfnl committed:
Merge pull request #20292 from Yoast/JRF/CS/static-closures

CS/QA: static closures should be declared as static

jrfnl committed:
Merge pull request #20293 from Yoast/JRF/QA/use-php-native-alternative

CS/QA: use the appropriate PHP native function

jrfnl committed:
Merge pull request #20294 from Yoast/JRF/CS/use-self

CS/QA: use `self` when refering to current class

jrfnl committed:
Merge pull request #20295 from Yoast/JRF/QA/fix-various-unused-undeclared-var-issues

QA: fix various unused/undeclared variable issues

jrfnl committed:
Merge pull request #20297 from Yoast/JRF/QA/appropriate-fqn-use

CS/QA; use fully qualified function calls/constants in the right places

jrfnl committed:
Merge pull request #20298 from Yoast/JRF/Docs/various-docs-fixes

Docs: various fixes

jrfnl committed:
Merge pull request #20299 from Yoast/JRF/CSQA/various-minor-fixes

CS/QA: various minor fixes

jrfnl committed:
Merge pull request #20296 from Yoast/JRF/QA/update-some-ignore-annotations

CS/QA: update some ignore annotations

jrfnl committed:
Merge pull request #20300 from Yoast/JRF/QA/dont-pass-non-existent-param

CS/QA: don't pass parameter which doesn't exist

vraja-pro committed:
Updated the description of the toggle.

pls78 committed:
Merge branch 'feature/add-dashboard-wincher-connect' of https://github.com/wincher-ab/wordpress-seo into add-dashboard-wincher-connect-patch

vraja-pro committed:
fixed quotation marks

leonidasmi committed:
Merge pull request #20286 from Yoast/20103-tighten-crawl-optimization-site-search-blocking-rules

20103 tighten crawl optimization site search blocking rules

pls78 committed:
Merge branch 'trunk' into add-dashboard-wincher-connect-patch

pls78 committed:
Fix shortlinks

leonidasmi committed:
Revert "Merge pull request #20268 from Yoast/revert/cron-indexation"

This reverts commit 8d8ba3636388905a825a961ca885b657a6838cd6, reversing
changes made to b722cce4b8b8d5ad53c5fd312332522f32aad249.

marinakoleva committed:
fixing a failing fulltexttest after incorrect decision during merge with trunk

FAMarfuaty committed:
Merge branch 'feature/html-parser' of github.com:Yoast/wordpress-seo into 20231-html-parser-token-positions-incorrect-when-sentence-starts-with-html-tag

vraja-pro committed:
Merge pull request #20301 from Yoast/JRF/Tests/fix-tests-not-running-and-bugfix

SEO_Links_Repository: fix tests not running and bugfix

igorschoester committed:
Merge pull request #20305 from Yoast/20304-revert-the-revert-of-the-background-cron-indexation

Background cron indexation

vraja-pro committed:
removed redundant assertions

hdvos committed:
remove unused argument in getTextElementPositions.js

vraja-pro committed:
Merge pull request #20302 from Yoast/JRF/Tests/add-missing-assertions

Tests: add missing assertions

vraja-pro committed:
Merge branch 'trunk' into 20271-fix-incorrect-covers-tags

FAMarfuaty committed:
Merge pull request #20246 from Yoast/20231-html-parser-token-positions-incorrect-when-sentence-starts-with-html-tag

20231 html parser token positions incorrect when sentence starts with html tag

vraja-pro committed:
Merge pull request #20285 from Yoast/20271-fix-incorrect-covers-tags

20271 fix incorrect covers tags

vraja-pro committed:
Merge pull request #20303 from Yoast/JRF/QA/fix-inconsistent-return-types

CS/QA: fix inconsistent return types

leonidasmi committed:
Merge branch 'trunk' into 20253-performance-run-get_limited_filtered_unindexed_count_background-only-in-selected-cases

pls78 committed:
Avoid an uncaught error when there's no network connection and the user clicks on the connect button

jrfnl committed:
GH Actions: merge unit and integration test workflows

... into one workflow, which still has the two separate jobs.

Includes updating the status badge in the `README`.

jrfnl committed:
CI/QA: start recording code coverage

This commit makes the necessary changes to start recording and uploading code coverage data to Coveralls.

Includes:
* Updating the GH Actions `test` workflow to run the high/low PHP version builds with code coverage and upload the results.
* Updating the PHPUnit config to record coverage and show an inline code coverage summary.
* Ignoring the potentially generated code coverage files in the `build/*` directory to prevent it being committed.
* Adding scripts to the `composer.json` file to run code coverage.
* Adding a code coverage badge to the README.

Notes:
* Coverage will be recorded using `Xdebug` as the code coverage driver.
* Recording code coverage can be slow and it is not needed to run it on each build to still get a good idea of the code coverage.
    With that in mind, code coverage is only recorded for high/low PHP.
* The `php-coveralls/php-coveralls` package is used to convert the coverage information from a format as generated by PHPUnit to a format as can be consumed by Coveralls.
    - As this package is only needed for the upload to Coveralls, the package has **not** been added it to the `composer.json` file, but will be (global) installed in the GH Actions workflow only.
    - The php-coveralls/php-coveralls package is not 100% compatible with PHP 8.x (yet), so for uploading the coverage generated using PHP 8.x, we normally switch to PHP 7.4 for uploading the code coverage reports, but you can only switch PHP version once per job and as we already do this for building the container, we cannot do it again.
         The chance of hitting the PHP 8.x problem is small anyway, so we'll just have to risk it for now.
* Coveralls requires a token to identify the repo and prevent unauthorized uploads.
    This token has been added to the repository secrets.
    People with admin access to the GH repo automatically also have access to the admin settings in Coveralls.
    If ever needed, the Coveralls token can be found (and regenerated) in the Coveralls admin for a repo.
    After regeneration, the token as stored in the GH repo Settings -> Secrets and Variables -> Actions -> Repository secrets should be updated.
* As the workflow sends multiple code coverage reports to Coveralls, we need to tell it to process those reports in parallel and give each report a unique name.
    That's what the `COVERALLS_PARALLEL` and `COVERALLS_FLAG_NAME` settings are about.
    This is also why the Clover files for the unit and the integration tests in the PHPUnit `logging` section have different names.
    Additionally, it is why the name for the Clover file for the multi-site run in the integration tests is overruled in the workflow. This allows us to save the coverage results for the single site and the multi site test runs in separate files and to upload both files to Coveralls for a more complete code coverage picture.
* The `coveralls-finish` job will signal to Coveralls that all reports have been uploaded.
    This basically gives the "okay" to Coveralls to report on the changes in code coverage via a comment on a GitHub PR as well as via a status check.

The pertinent Coveralls settings used for this repo are:
* "Only send aggregate Coverage updates to SCM": enabled
* "Leave comments": enabled
* "(Comment) Format": detailed
* "Use Status API": enabled
* "Coverage threshold for failure": 36%
* "Coverage decrease threshold for failure": 1.0%

jrfnl committed:
Merge pull request #20309 from Yoast/JRF/ghactions-merge-test-workflows

GH Actions: merge unit and integration test workflows

jrfnl committed:
Merge pull request #20310 from Yoast/JRF/start-recording-code-coverage

CI/QA: start recording code coverage

jrfnl committed:
Merge pull request #20311 from Yoast/JRF/phpcs-eslint-lower-thresholds

CS/QA: lower thresholds

jrfnl committed:
GH Actions: run workflows more selectively

Only run the workflows when there is a chance that the result of the workflow run will be different from before.

The path selection is set up to cast a wide net to make the risk of the workflows not running when the build could potentially break as small as possible.

This means that, aside from files of the type being checked in the workflow changing, the workflow will also run when:
* The configuration used changes.
* Dependencies used in the workflow may potentially have changed.
* Composer scripts used in the workflow may potentially have changed.
* Shell scripts used in the workflow have changed.
* The workflow itself changes.
* Miscellaneous related files have changed.

Note: the lists of files _may_ include some files which don't currently exist in the repo (like `.eslintignore`) for consistency across workflows and to prevent issues if/when such file(s) would be introduced to the repo.

diedexx committed:
Merge pull request #20312 from Yoast/JRF/QA/indexable-presentations-clean-up-properties

thijsoo committed:
Merge pull request #20257 from Yoast/20253-performance-run-get_limited_filtered_unindexed_count_background-only-in-selected-cases

KaisZaoualiWincher committed:
Fix WincherPerformanceReport warnings

pls78 committed:
Merge branch 'feature/add-dashboard-wincher-connect' of https://github.com/wincher-ab/wordpress-seo into add-dashboard-wincher-connect-patch

vraja-pro committed:
Merge pull request #20307 from Yoast/add-dashboard-wincher-connect-patch

Add dashboard wincher connect patch

enricobattocchi committed:
Prevents metabox crashing when the additional keywords objects are malformed

enricobattocchi committed:
Add release text and removes mentions to Zapier

mhkuu committed:
Adds the inclusive language score to WPSEO_Rank

mhkuu committed:
Allows overriding the applicable filters and the order by part of the query, also turns two methods static to allow easy re-use

FAMarfuaty committed:
Merge branch 'trunk' of github.com:Yoast/wordpress-seo into feature/html-parser

mhkuu committed:
Adds tests for WPSEO_Rank

mhkuu committed:
Reverts making is_valid_filter static

vraja-pro committed:
Merge pull request #20336 from Yoast/fix-wincher-crash-on-malformed-focus-keywords

Prevents metabox crashing when the additional keywords objects are malformed

enricobattocchi committed:
Update deprecation version to 20.10

mhkuu committed:
Apply suggestions from code review

Co-authored-by: Enrico Battocchi <enrico.battocchi@gmail.com>

mhkuu committed:
Reverts making generate_option static

leonidasmi committed:
Add new setting for AdsBot crawling

leonidasmi committed:
Fix JS linting

leonidasmi committed:
Uses the new setting to actually add AdsBot disallow rules in robots.txt

agnieszkaszuba committed:
Merge pull request #20338 from Yoast/add-inclusive-language-filter

Enables creating filters for other scores

vraja-pro committed:
Merge pull request #20284 from Yoast/20242-phase-1-remove-usage-of-yoast-components

Remove usage of yoast-components

leonidasmi committed:
Merge branch 'trunk' into 3881-add-a-crawl-cleanup-toggle-to-prevent-adsbot-from-crawling

Commit:
Bump version to 20.8

Commit:
Merge branch 'release/20.8'

Commit:
Bump version to 20.8 on free

mhkuu committed:
Merge branch 'main' of github.com:Yoast/wordpress-seo into trunk

leonidasmi committed:
Update @covers tags

mhkuu committed:
Merge branch 'trunk' of github.com:Yoast/wordpress-seo into feature/html-parser

Commit:
Add changelog

Commit:
Bump Gutenberg supported version.

Commit:
Bump version to 20.9-RC1

leonidasmi committed:
Improve changelog

leonidasmi committed:
Empty-Commit

leonidasmi committed:
Merge pull request #20351 from Yoast/add-to-20-9-changelog

Add to 20.9 changelog

Commit:
Add changelog

Commit:
Bump version to 20.9-RC2

noud-github committed:
Merge pull request #20316 from Yoast/JRF/ghactions-run-more-selectively

GH Actions: run workflows more selectively

agnieszkaszuba committed:
Update README.md

Updates a table that mentions English as the only language that has morphology support. Now more languages have morphology support.

hdvos committed:
fix mistake where it was impossible to undo highlighting.

hdvos committed:
remove irrelevant spec in markTinyMCE.test.js

marinakoleva committed:
Merge branch 'trunk' of https://github.com/Yoast/wordpress-seo into feature/html-parser

marinakoleva committed:
Merge branch 'feature/html-parser' of https://github.com/Yoast/wordpress-seo into 20356-html-parser-fix-removing-highlighting-classic-editor

leonidasmi committed:
Don't add terms with custom canonical URLs in the sitemap

leonidasmi committed:
Add back old string check

agnieszkaszuba committed:
Merge pull request #20358 from Yoast/20356-html-parser-fix-removing-highlighting-classic-editor

fix mistake where it was impossible to undo highlighting.

thijsoo committed:
Merge pull request #20340 from Yoast/3881-add-a-crawl-cleanup-toggle-to-prevent-adsbot-from-crawling

leonidasmi committed:
Don't add sitemap rewrite rules when Yoast SEO is being deactivated

enricobattocchi committed:
Merge pull request #20361 from Yoast/20126-taxonomies-with-custom-canonical-urls-should-be-excluded-from-xml-sitemaps

Don't add terms with custom canonical URLs in the sitemap

mhkuu committed:
Merge pull request #20367 from Yoast/feature/html-parser

Introduces HTML processing for some SEO assessments

leonidasmi committed:
Merge branch 'trunk' into 16459-deactivating-plugin-does-not-seem-to-flush-rewrite-rules

leonidasmi committed:
Prevent all hooks running when on deactivation

leonidasmi committed:
Unslash the GET parameters when checking if we are on deactivating the plugin

leonidasmi committed:
Sanitize the GET parameters now that we use them

leonidasmi committed:
Fix test description

enricobattocchi committed:
Update src/conditionals/deactivating-yoast-seo-conditional.php

enricobattocchi committed:
Merge pull request #20368 from Yoast/16459-deactivating-plugin-does-not-seem-to-flush-rewrite-rules

Don't add sitemap rewrite rules when Yoast SEO is being deactivated

vraja-pro committed:
Merge pull request #19883 from Yoast/remove-old-premium-notification

Deprecate the integration to display a notification when Premium is not up-to-date

KaisZaoualiWincher committed:
Fix returning empty array instead of empty object in Wincher tracking keyword related actions
